packages/react/select/src/Select.tsx

```diff
@@ -624,7 +625,25 @@ const SelectContentImpl = React.forwardRef<SelectContentImplElement, SelectConte
       }
     });
 
+    const handleItemSeen = React.useCallback(
+      (node: SelectItemElement | null, value: string, disabled: boolean) => {
+        const isFirstSeenValidItem = !firstItemRef.current && !disabled;
+        const isSelectedItem = context.value !== undefined && context.value === value;
+        if (isSelectedItem || isFirstSeenValidItem) {
+          setSelectedItem(node);
+          if (isFirstSeenValidItem) firstItemRef.current = true;
+        }
+      },
+      [context.value]
+    );
     const handleItemLeave = React.useCallback(() => content?.focus(), [content]);
+    const handleItemTextSeen = React.useCallback(
```

Comment:
I'm not sure about usage of `seen` in the naming, I feel like we should be more specific as relates to the internals. Struggling for alternatives though but is it right to say we want to convey this handles when items are "available" or have transitioned into "actionable" / "rendered" state?

Reply:
I'm not sure I follow here.

Reply:
What does “seen” refer to in this context?

Reply:
I wanted to say it's been seen by the "content" part, as in it knows it mounted through the callback ref.
It's making me question now though where I need to guard it with `if (node)` too? 🤔

Reply:
Gotcha, is it clearer to keep it simple `handleItemMount`? I don't feel super strongly tbh, just that "seen" makes me think of visibility logic or something with overflow vs lifecycle.

> It's making me question now though where I need to guard it with if (node) too?

hmm, not sure it makes much difference, `position` still has to guard anyway right?